tests/test_checker.py

```diff
@@ -226,3 +232,25 @@ def test_sbomchecker_output_html():
     )
 
     assert got == expected
+
+def test_components_without_functions():
+    logger = logging.getLogger(__name__)
+    logger.info("In test")
+    filepath = os.path.join(
+        os.path.dirname(__file__), "data", "other_tests", "test_components_without_functions.spdx"
+    )
+    sbom = sbom_checker.SbomChecker(filepath)
+    components = sbom.get_components_without_names()
+    assert components == ["SPDXRef-Package1"]
+    components = sbom.get_components_without_versions()
+    assert components == ["glibc-no-version-1", "glibc-no-version-2"]
+    components = sbom.get_components_without_versions(returnTuples=True)
+    assert components == [("glibc-no-version-1", "SPDXRef-Package2"), 
+                          ("glibc-no-version-2", "SPDXRef-Package3")]
+    components = sbom.get_components_without_suppliers()
+    assert components == ["glibc-no-supplier"]
+    components = sbom.get_components_without_suppliers(returnTuples=True)
+    assert components == [("glibc-no-supplier", "SPDXRef-Package4")]
+    # TODO: Not sure how to test this. If any package misses the SPDXID the whole file seems to be invalid.
+    #components = sbom.get_components_without_identifiers()
+    #assert components == ["glibc-no-identifier"]
```

Comment:
Yeah, this is un-testable, I also think, given the constraints of the `spdx-tools` python package that powers this project. I've run into this "problem" too. Since I think that's the case, it's probably just worth explaining this situation in the comment directly to help aid future developers.